### DISCUSSION TITLE
x-trap.inert in the focus plugin causes undesirable scroll to the bottom

### DISCUSSION THREAD

**Discussion by C-alcium** [alpinejs/alpine#3159] [7. Bugs (archived)]
If you have an element that you toggle on and off with x-show

```html
<div x-show="open" ... >
    <div class="large-scrollable-content" ... > ... </div>
</div>
```

and the content is very large, and has y-overflow: auto, setting `x-trap.inert.noscroll="show"` will result in the content always being scrolled to the bottom when `open` is set to `true`. 

I believe this is because of the focus plugin's use of `el.focus`, although I could be wrong. It seems like the solution would be to get the first element inside the element that has content that can be focused on, and focusing on that (e.g. some text), but I'm not sure. I just wanted to put this in here because it seemed like no one else was having this issue.

**Comment by aaronstezycki** [comment]
I've got this issue currently. I'm opening a modal, and the modal contains buttons at the bottom of some scrollable content. When the modal opens, the focus is directly set on the buttons and scrolls the content to the bottom to make that happen. I'm just using default `x-trap.noscroll` on the container.

This obviously isn't desirable behaviour.

**Comment by aaronstezycki** [comment]
Just looking at the focus page on Alpine.js, and it seems none of the latter examples work. Either the focus trapping or the focus styles for the buttons do not work.

[https://alpinejs.dev/plugins/focus](https://alpinejs.dev/plugins/focus)

**Comment by aaronstezycki** [comment]
Also confirmed on the modal AlpineJS modal component. [Alpine Modal](https://alpinejs.dev/component/modal)

Simply stuffing content into the modal so it has some vertical scrolling (can do this via the inspector), and then open the modal and it will automatically focus the buttons at the bottom.

**Comment by aaronstezycki** [comment]
The only way I can think for rudimentary fix for this for time being is setting tabindex="1" on the first element in the modal causing the auto focus to scope that particular element instead of buttons located at the bottom of the modal.

**Reply by ekwoka** [reply]
You could just put a close X at the top or something

**Comment by stepanjakl** [comment]
Hi, I've been experiencing a similar issue, which is also linked to a custom modal component. The problem arises when the content scrolls down since there's a focusable button positioned at the bottom. I believe it would be beneficial to introduce a modifier, such as `.nofirstfocus`, to disable focus when the x-trap directive/attribute is triggered.

**Comment by rosswintle** [comment]
I popped in here with the same question. Looks like [v3.13.4](https://github.com/alpinejs/alpine/releases/tag/v3.13.4) added a new `noautofocus` modifier for `x-trap` that should help.

See https://github.com/alpinejs/alpine/pull/3977 and the [documentation for the modifier](https://alpinejs.dev/plugins/focus#noautofocus)
